src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -64,13 +64,13 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
                     continue;
                 }
 
-                if (node.TagHelper.IsBindTagHelper() && node.AttributeName.StartsWith("bind"))
+                if (node.TagHelper.IsBindTagHelper() && node.IsDirectiveAttribute && node.AttributeName.StartsWith("bind"))
```

Comment:
Is there a reason why we need the `StartsWith` check here? What is that actually doing?

Reply:
It is no longer needed. Will remove.